src/usr/share/netcheck/netcheck.py

```diff
@@ -70,37 +68,40 @@ def _dns_query(self, network, nameserver, query_name):
         """
         # TODO: Use something more secure than unauthenticated plaintext DNS requests.
 
-        self.logger.trace('Querying %s for %s on network %s.' % (nameserver, query_name, network))
+        self.logger.trace('Querying %s for %s on network %s.', nameserver, query_name, network)
         success = False
 
-        # TODO: Add support for when this is not able to obtain the IP.
-        interface_ip = self.network_meta.get_interface_ip(network)
-
-        self.resolver.nameservers = [nameserver]
-        try:
-            query_result = self.resolver.query(query_name, source=interface_ip)
-            success = True
-
-        except dns.resolver.Timeout as detail:
-            # Network probably disconnected.
-            self.logger.error('DNS query for %s from nameserver %s on network %s timed out. %s: %s' %
-                (query_name, nameserver, network, type(detail).__name__, detail.message))
+        interface_ip = self.network_helper.get_connection_ip(network)
```

Comment:
I'd probably go ahead and change most mentions of 'network' to 'connection'. I went through the code and it doesn't seem like all that much work.

Reply:
"Doesn't seem like all that much work."

My 37k diff says otherwise.